src/Microsoft.VisualStudio.LanguageServices.Razor/DefaultTagHelperCompletionFactsService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    [Export(typeof(TagHelperCompletionFactsService))]
+    internal class DefaultTagHelperCompletionFactsService : TagHelperCompletionFactsService
+    {
+        [Import(typeof(TagHelperFactsService))]
+        public TagHelperFactsService TagHelperFactsService { get; set; }
+
+        public override ElementCompletionResult GetElementCompletions(
+            TagHelperDocumentContext documentContext,
+            IEnumerable<string> existingCompletions,
+            string containingTagName,
+            IEnumerable<KeyValuePair<string, string>> attributes,
+            string containingParentTagName,
+            Func<string, bool> inHTMLSchema)
```

Comment:
I think that all of this stuff needs to be rolled up into some kind of context object.